docs/source/architecture.rst

```diff
@@ -4,14 +4,62 @@
 Architecture of nbconvert
 =========================
 
-This is a high-level outline of the structure and objects in nbconvert,
-and how they are used in the pipeline of converting a notebook to any given format.
-
+This is a high-level outline of the basic workflow, structures and objects in nbconvert.
+Specifically, this exposition has a two-fold goal: 
+    
+    #. to alert you to the affordances available for customisation or direct contributions 
+    #. to provide a map of where and when different events occur, which should aid in tracking down bugs.
+    
+
+A detailed pipeline exploration
+===============================
+
+Nbconvert takes in a notebook, which is a JSON object, and operates on that object. 
+
+This can include operations that take a notebook and return a notebook.
+For example, that operation could be to execute the notebook as though it were a continuous script; if it were executed ``--in-place`` then it would overwite the current notebook.
+Or it could be that we wish to systematically alter the notebook, for example by clearing all output cells.
+Format agnostic operations on cell content that do not violate the nbformat spec can be interpreted as a notebook to notebook conversion step; such operations can be performed as part of the preprocessing step.
+
+But often we want to have the notebook's structured content in a different format.
+Importantly, in many cases the structure of the notebook should be reflected in the structure of the output, adapted to the output's format.
+For that purpose, the original JSON structure of the document is crucial scaffolding needed to support this kind of structured output.
+In order to maintain structured, it can be useful to apply our conversion programmatically on the structure itself.
+To do so, when converting to formats other than the notebook, we use the `jinja`_ templating engine.
```

Comment:
Jinja is not mandatory, one could have an exporter that does not use Jinja , we might want to slight rephrase that maybe.